### PR TITLE
Task management v1.2

### LINKED ISSUE #96: Register the project on google developer console and enable google tasks api

### LINKED ISSUE #97: Authorize access to google tasks

### LINKED ISSUE #98: Add a new task list to user google account

### LINKED ISSUE #99: Customize Google Tasks API

### PR DESCRIPTION
**Pull request by EdwardKSG**
the beginning step of major enhancement (task management).
1. register project, get api client credentials close #96
2. write program for api connection close #97
3. add a few methods in the modeling part close #99
4. add a new command (incomplete) which create a new default task list close #98
PS: currently the software will open browser for user to login with google account. later will load the webpage in the app when I work on UI part.

### CODE REVIEW

File: src/main/java/seedu/progresschecker/logic/commands/AddDefaultTasksCommand.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+     */
+    public AddDefaultTasksCommand(String title) {
+        requireNonNull(title);
+        listTitle = title;
+    }
+
+    @Override
+    public CommandResult execute() throws CommandException {
+        try {
+            createTaskList(listTitle);
```
Comment (kohyeeru) [inline src/main/java/seedu/progresschecker/logic/commands/AddDefaultTasksCommand.java:40]: I understand that you mentioned this command is incomplete so I'd just like to clarify on the behavior. Right now this statement causes some NoClassDefFoundError and/or ClassNotFoundException (I think), is this behavior intended? If yes, would you be completing this command by the end of milestone v1.3? o:

File: src/main/java/seedu/progresschecker/model/person/Person.java [hunk 2]
```diff
@@ -10,7 +10,7 @@
 import seedu.progresschecker.model.tag.UniqueTagList;
 
 /**
- * Represents a Person in the major book.
+ * Represents a Person in the progressChecker.
```
Comment (kohyeeru) [inline src/main/java/seedu/progresschecker/model/person/Person.java]: "**P**rogressChecker"? :>

**Review by kohyeeru — CHANGES_REQUESTED** [review]
Great that you included header comments for the methods and made use of exception catching! I only have some clarification with the expected behavior of the app:

1) Had the impression that there will be some sort of login prompt on software start up from your PR comment but I didn't encounter any, was I supposed to see anything? o:

2) Is adding of new task list the command where a new user would run only once to add all pre-set LOs to their Google Tasks after authorization?

File: src/main/java/seedu/progresschecker/model/person/Person.java (hunk 2 above)
Comment (EdwardKSG) [reply src/main/java/seedu/progresschecker/model/person/Person.java]: yep

File: src/main/java/seedu/progresschecker/logic/commands/AddDefaultTasksCommand.java (hunk 1 above)
Comment (EdwardKSG) [reply src/main/java/seedu/progresschecker/logic/commands/AddDefaultTasksCommand.java:40]: updated

**Review by kohyeeru — APPROVED** [review]

**Pull request merged**